onlinejudge/_implementation/command/submit.py

```diff
@@ -142,7 +142,7 @@ def submit(args: 'argparse.Namespace') -> None:
         # show result
         if args.open:
             browser = webbrowser.get()
-            log.status('open the submission page with: %s', browser.name)
+            log.status('open the submission page with: %s', browser.name if hasattr(browser, 'name') else browser._name)
```

Comment:
```suggestion
            log.status('open the submission page with: %s', browser.name if hasattr(browser, 'name') else browser._name)  # type: ignore
```
I know this is not good solution, but I don't have any other idea.

Reply:
I found that we can suppress the following error:
```error: "BaseBrowser" has no attribute "_name"; maybe "name"?```
by replacing `hasattr(...)` to `not isinstance(browser, webbrowser.MacOSXOSAScript)`.

This method can tell `mypy` that the browser object except `MacOSXOSAScript` is not assigned to the else statement. However, this modification raised another error:
```error: "MacOSXOSAScript" has no attribute "_name"```

I think this is clearly the bug of the type definition even the behavior "only the `MacOSXOSAScript` has `._name` property instead of `.name`" is not a bug. We can find that there is no definition for `MacOSXOSAScript._name` property in the typeshed official repository.
https://github.com/python/typeshed/blob/616cb0bb656d6070d9c2af4185c7922b2af134c6/stdlib/2and3/webbrowser.pyi#L96-L98

I felt that your proposal is reasonable in such a situation, or creating own `.pyi` file also seems to be a good option.

Reply:
Thank you for your investigation.
`.name` is not public API. https://docs.python.org/ja/3.7/library/webbrowser.html does not say about `.name`.

How about `log.status('open the submission page with browser.)`?

Benefit of specifying browser name is small for user and developer, and `.name` may not be supported in the future Python version. `isinstance(browser, webbrowser.MacOSXOSAScript)`branch and `.pyi` can increase maintenance work.

Reply:
Thank you for your proposal. I followed that (please see 7f0b498).

Reply:
Thanks, at least, `_name` is not published API. Since it is starts with `_`. We should not use it.